src/components/Tabs/Annotation/AnnotationTrackThumbnail.tsx

```diff
@@ -112,30 +178,43 @@ export default function AnnotationTrackThumbnail(inputProps: AnnotationTrackThum
       return;
     }
 
-    const drawDashedLine = (x: number, color: string): void => {
-      ctx.strokeStyle = color;
-      ctx.setLineDash([2, 2]);
-      ctx.beginPath();
-      ctx.moveTo(x, 0);
-      ctx.lineTo(x, props.heightPx);
-      ctx.stroke();
-      ctx.closePath();
-    };
-
     // Draw hovered time on canvas.
     if (props.frame && hoveredCanvasX !== null) {
       const hoveredTime = xCoordToTime(hoveredCanvasX);
       const x = timeToXCoord(hoveredTime + 0.5);
-      drawDashedLine(x, theme.color.text.hint);
+      drawDashedLine(ctx, x, props.heightPx, theme.color.text.hint);
     }
     // Draw the current time on the canvas. Replace with awaiting time if it is
     // set for optimistic updates
     if (props.frame !== undefined && props.frame >= minTime && props.frame <= maxTime) {
       const frame = awaitingFrame !== null ? awaitingFrame : props.frame;
       const x = timeToXCoord(frame + 0.5);
-      drawDashedLine(x, theme.color.text.primary);
+      drawDashedLine(ctx, x, props.heightPx, theme.color.text.primary);
+    }
+
+    if (props.mark !== undefined) {
+      const x = timeToXCoord(props.mark + 0.5);
+      drawMark(ctx, x, theme.color.text.primary);
+    }
+  }, [track, dataset, props.mark, props.widthPx, props.heightPx, props.frame, hoveredCanvasX, awaitingFrame]);
+
+  // Update highlight canvas
+  useEffect(() => {
+    const canvas = highlightCanvasRef.current;
+    const ctx = highlightCanvasRef.current?.getContext("2d");
+    if (!canvas || !ctx) {
+      return;
     }
-  }, [track, dataset, props.widthPx, props.heightPx, props.frame, hoveredCanvasX, awaitingFrame]);
+    ctx.clearRect(0, 0, canvas.width, canvas.height);
+    if (props.highlightedIds && dataset) {
+      const highlightedTimes = props.highlightedIds.map((id) => dataset.getTime(id));
+      const highlightIntervals = getIntervals(highlightedTimes);
+      for (const interval of highlightIntervals) {
+        const xInterval: [number, number] = [timeToXCoord(interval[0]), timeToXCoord(interval[1] + 1)];
+        drawInterval(ctx, xInterval, props.heightPx, "#ebc88d9f");
```

Comment:
nit: name this color?

Reply:
Whoops, missed this. I'll make this a defined color in the app style